activerecord/lib/active_record/store.rb

```diff
@@ -290,7 +290,7 @@ def self.as_indifferent_hash(obj)
           def as_regular_hash(obj)
             case obj
             when ActiveSupport::HashWithIndifferentAccess
-              obj.to_h
+              obj.to_hash
             when Hash
               obj
```

Comment:
Maybe this is too contrived to consider, but technically you could have a `Hash` with a `HashWithIndifferentAccess` nested inside

Reply:
Great point @tkalliom. Looks like it would have to be a recursive solution.